### PR TITLE
fix: allow correcting admin email during first-time setup (ATT-265)

### PR DESCRIPTION
**Pull request by jappyjan**
## Summary

Fixes ATT-265: When an admin enters a wrong email during first-time setup, they get permanently locked out — the verification email goes to the wrong address, login requires a verified email, and setup can't be redone.

- Add unauthenticated `POST /users/correct-setup-email` endpoint that allows changing the admin email when setup is incomplete (exactly 1 user, email not verified, valid credentials required)
- Add "Wrong email?" correction form to the frontend VerifyEmailStep with EN/DE translations
- 13 exhaustive tests covering all guard conditions, error propagation, and side-effect isolation

## Test plan

- [x] All 647 API tests pass (including 13 new `correctSetupEmail` tests)
- [x] Frontend typecheck passes
- [x] Frontend builds successfully
- [x] Pre-commit hooks pass (lint, typecheck, build, test)
- [ ] Manual: complete first-time setup with wrong email, use correction form, verify new email received

Closes #556

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Allow correction of the initial admin email during first-time setup when the original address was entered incorrectly, and surface this flow in the first-time setup UI.

New Features:
- Introduce an unauthenticated POST /users/correct-setup-email endpoint to let admins update their email during incomplete first-time setup with username/password verification.
- Add a first-time setup UI flow that lets admins correct their email address and resend the verification email from the Verify Email step.

Bug Fixes:
- Prevent admins from being permanently locked out when an incorrect email is entered during first-time setup by enabling secure email correction and verification resend.

Enhancements:
- Extend shared API client types, schemas, and React Query hooks to support the new correct-setup-email operation.
- Add comprehensive controller tests covering all guard conditions, error propagation, and side effects for the new email correction endpoint.

Tests:
- Add a suite of tests for the correctSetupEmail controller logic to cover success, validation, authorization, and edge cases.

### CODE REVIEW

File: apps/api/src/users-and-auth/users/users.controller.ts [hunk 1]
```diff
@@ -444,6 +446,66 @@ export class UsersController {
...
+        message: { type: 'string', example: 'Email updated and verification email sent' },
+      },
+    },
+  })
+  @ApiResponse({ status: 403, description: 'First-time setup is already complete.' })
+  @ApiResponse({ status: 401, description: 'Invalid username or password.' })
+  async correctSetupEmail(@Body() body: CorrectSetupEmailDto) {
+    this.logger.debug(`Correcting setup email for username: ${body.username}`);
+
+    const admin = await this.usersService.findOne({ username: body.username });
```
Comment (jappyjan) [inline apps/api/src/users-and-auth/users/users.controller.ts]: Can be abused to scan for all registered users.

First check if it's still first time setup and fail early on that instead of first failing on wrong username/email

File: apps/api/src/users-and-auth/users/users.controller.ts [hunk 2]
```diff
@@ -444,6 +446,66 @@ export class UsersController {
...
+    if (!admin) {
```
Comment (jappyjan) [inline apps/api/src/users-and-auth/users/users.controller.ts]: We should not only allow email change but also password, username etc. With every correction request we should delete the existing non verified user and recreate it fully with all data. Maybe we can also just reuse the existing first time user creation endpoint with a "overwrite" flag or so instead of a second correction endpoint.

**Comment by jappyjan** [comment]
@claude please fix all comments
